### PR TITLE
WIP: CRD and Operator pattern

### PR DESCRIPTION
**Pull request by ejether**
* adding custom resource definition, custom resource example and proper rbac to access the crd and cr

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,9 +1,9 @@
-FROM python:3.6.4-alpine3.7
+FROM python:2.7.14-alpine3.7
```
Comment (robscott) [inline Dockerfile]: Any reason for moving to Python 2?

File: manage_rbac.py [hunk 2]
```diff
@@ -174,43 +199,83 @@ def __init__(self, rbac_users={}):
...
-    parser.add_argument('--config', help='YAML configuration file to load', required=True)
+    parser.add_argument('--config', help='YAML configuration file to load')
     parser.add_argument('--kubectl-auth', action='store_true', help='Use kubectl command to refresh auth (useful for GKE)')
     args = parser.parse_args()
     if args.kubectl_auth:
         os.system('kubectl get ns >/dev/null 2>&1')
-    RBACManager(yaml.load(open(args.config)))
+    try:
+        if args.config is not None:
+            RBACManager().update(file=args.config)
```
Comment (robscott) [inline manage_rbac.py]: It seems like some logging here and on the line below to explain what mode this is running in would be helpful.

File: Dockerfile (hunk 1 above)
Comment (ejether) [reply Dockerfile]: The new version doesn't work with python3.

File: manage_rbac.py (hunk 2 above)
Comment (ejether) [reply manage_rbac.py]: 👍

**Comment by robscott** [comment]
I've made some changes to the k8s examples and readme, both the job and controller pattern are working well for me now.

**Comment by ejether** [comment]
Looks good. Ready to merge?

**Pull request merged**